src/PerformanceSensitiveAnalyzers/CSharp/AvoidAllocationWithArrayEmptyCodeFix.cs

```diff
@@ -0,0 +1,287 @@
+                                _title);
+                            context.RegisterCodeFix(codeAction, diagnostic);
+                        }
+                    }
+                    break;
+            }
+        }
+
+
+        private static bool IsMethodInvocationParameter(SyntaxNode node) => node is ArgumentSyntax;
+
+        private static bool IsReturnStatement(SyntaxNode node)
+        {
+            return node.Parent is ReturnStatementSyntax || node.Parent is YieldStatementSyntax || node.Parent is ArrowExpressionClauseSyntax;
+        }
+
+        private static bool IsInsideMemberReturningEnumerable(SyntaxNode node, SemanticModel semanticModel)
+        {
+            return IsInsideMethodReturningEnumerable(node, semanticModel) ||
+                   IsInsidePropertyDeclaration(node, semanticModel);
+
+        }
+
+        private static bool IsInsidePropertyDeclaration(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var propertyDeclaration = node.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            if (propertyDeclaration != null && IsPropertyTypeReadonlySequence(semanticModel, propertyDeclaration))
+            {
+                return IsAutoPropertyWithGetter(node) || IsArrowExpression(node);
+            }
+
+            return false;
+        }
+
+        private static bool IsAutoPropertyWithGetter(SyntaxNode node)
+        {
+            var accessorDeclaration = node.FirstAncestorOrSelf<AccessorDeclarationSyntax>();
+            return accessorDeclaration != null && accessorDeclaration.Keyword.Text == "get";
+        }
+
+        private static bool IsArrowExpression(SyntaxNode node)
+        {
+            return node.FirstAncestorOrSelf<ArrowExpressionClauseSyntax>() != null;
+        }
+
+        private static bool CanBeReplaceWithEnumerableEmpty(ArrayCreationExpressionSyntax arrayCreation)
+        {
+            return IsInitializationBlockEmpty(arrayCreation.Initializer);
+        }
+
+        private static bool CanBeReplaceWithEnumerableEmpty(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        {
+            return IsCollectionType(semanticModel, objectCreation) &&
+                   IsInitializationBlockEmpty(objectCreation.Initializer) &&
+                   IsCopyConstructor(semanticModel, objectCreation) == false;
+        }
+
+        private static bool IsInsideMethodReturningEnumerable(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var methodDeclaration = node.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            return methodDeclaration != null && IsReturnTypeReadonlySequence(semanticModel, methodDeclaration);
+        }
+
+        private static async Task<Document> Transform(Document contextDocument, SyntaxNode node, TypeSyntax typeArgument, CancellationToken cancellationToken)
+        {
+            var noAllocation = SyntaxFactory.ParseExpression($"Array.Empty<{typeArgument}>()");
```

Comment:
💡 This should generate a type syntax from the special System_Array type obtained from the compilation.